apps/ledger-live-mobile/src/components/FormatDate.tsx

```diff
@@ -1,33 +1,44 @@
 import React from "react";
 import { useSelector } from "react-redux";
+import { createSelector } from "reselect";
 import compareDate from "../logic/compareDate";
 import { localeSelector } from "../reducers/settings";
 
 type Props = {
   date: Date | null | undefined;
   withHoursMinutes?: boolean;
 };
-const defaultOptions: Intl.DateTimeFormatOptions = {
-  year: "numeric",
-  month: "numeric",
-  day: "numeric",
-};
-const hoursAndMinutesOptions: Intl.DateTimeFormatOptions = {
-  hour: "2-digit",
-  minute: "2-digit",
-};
+
+const defaultOptionsSelector = createSelector(
+  localeSelector,
+  locale =>
+    new Intl.DateTimeFormat(locale, {
+      year: "numeric",
+      month: "numeric",
+      day: "numeric",
+    }),
+);
+
+const hoursAndMinutesOptionsSelector = createSelector(
+  localeSelector,
+  locale =>
+    new Intl.DateTimeFormat(locale, {
+      hour: "2-digit",
+      minute: "2-digit",
+    }),
+);
 
 function FormatDate({
   date,
   withHoursMinutes = false,
 }: Props): JSX.Element | null {
-  const locale = useSelector(localeSelector);
+  const defaultOptions = useSelector(defaultOptionsSelector);
+  const hoursAndMinutesOptions = useSelector(hoursAndMinutesOptionsSelector);
   const jsx =
     date && date.getTime()
```

Comment:
Just asking for opinion here but shouldn't we instead have date be a mandatory prop
```ts
type Props = {
  date: Date;
  withHoursMinutes?: boolean;
};
```

And only render this component is date exists ?

```ts
date && <FormatDate date={date} />
```
This will avoid this check here but also this would avoid calling `createSelector` twice and then returning null

Reply:
Nice! Didn't know

Reply:
A simpler solution would have been to use the `useMemo` hook